sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/CheckpointPosition.cs

```diff
@@ -21,15 +21,40 @@ public struct CheckpointPosition : IEquatable<CheckpointPosition>
         ///
         public long SequenceNumber { get; }
 
+        /// <summary>
+        ///   The sequence number to associate with the checkpoint. This indicates that a processor should begin reading from the next event in the stream.
+        /// </summary>
+        ///
+        public string GlobalOffset { get; }
+
         /// <summary>
         ///   Initializes a new instance of the <see cref="CheckpointPosition"/> struct.
         /// </summary>
         ///
         /// <param name="sequenceNumber">The sequence number to associate with the checkpoint. This indicates that a processor should begin reading from the next event in the stream.</param>
         ///
+        /// <remarks>
+        ///   This constructor is not compatible when processing a geo-replicated Event Hub. Use <see cref="CheckpointPosition(string, long)"/> or
+        ///   <see cref="FromEvent(EventData)"/> instead.
+        /// </remarks>
+        ///
         public CheckpointPosition(long sequenceNumber)
         {
             SequenceNumber = sequenceNumber;
+            GlobalOffset = null;
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="CheckpointPosition"/> struct.
+        /// </summary>
+        ///
+        /// <param name="globalOffset">The global offset to associate with the checkpoint. This indicates that a processor should begin reading from the next event in the stream.</param>
+        /// <param name="sequenceNumber">The sequence number to associate with this checkpoint. This value is only used as informational metadata for the checkpoint.</param>
+        ///
+        public CheckpointPosition(string globalOffset, long sequenceNumber = default)
```

Comment:
`default` is 0 for this.   I think we'll want to default it to the min value or make it nullable.